app/api/helpers/ticketing.py

```diff
@@ -24,16 +25,16 @@ def get_order_expiry():
     @staticmethod
     def match_discount_quantity(discount_code, ticket_holders=None):
         qty = 0
-        old_holders = get_count(TicketHolder.query.filter(TicketHolder.ticket_id.in_(discount_code.tickets.split(","))))
-
+        ticket_ids = [ticket.id for ticket in discount_code.tickets]
+        old_holders = get_count(TicketHolder.query.filter(TicketHolder.ticket_id.in_(ticket_ids))
+                                .join(Order).filter(Order.status.in_(['completed', 'placed'])))
         for holder in ticket_holders:
             ticket_holder = TicketHolder.query.filter_by(id=holder).one()
-            if ticket_holder.ticket.id in discount_code.tickets.split(","):
+            if ticket_holder.ticket.id in ticket_ids:
                 qty += 1
         if (qty + old_holders) <= discount_code.tickets_number and \
```

Comment:
At this condition, what you could do is, suppose the maximum number of tickets on which the discount code can be applied is 3 and you are trying to order 5 tickets, apply the discount only on 3. This is how it's implemented on other platforms.